server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/pluggable_scm_material_representer.rb

```diff
@@ -20,7 +20,12 @@ module Materials
             class PluggableScmMaterialRepresenter < ApiV1::BaseRepresenter
                 alias_method :material_config, :represented
 
-                property :scm_id, as: :ref
+                property :scmId, as: :ref, setter: lambda { |value, options|
+                  scm = options[:config].getSCMs().find(value)
+                  self.setSCMConfig(scm) if(scm)
+                  self.setScmId(value)
+                }
```

Comment:
Should this  `raise ApiV1::UnprocessableEntity` error if the SCM is not found? —


```ruby
property :scmId, as: :ref, setter: lambda { |val, opts|
  if scm = options[:config].getSCMs().find(value)
    self.setSCMConfig(scm)
    self.setScmId(value
  else
    raise ApiV1::UnprocessableEntity, "A pluggable SCM with identifier `#{value}` was not found!"
  end
}
```